django_app/tests/test_views.py

```diff
@@ -371,6 +371,38 @@ def test_view_session_with_documents(chat_message: ChatMessage, client: Client):
     assert b"original_file.txt" in response.content
 
 
+@pytest.mark.django_db()
+def test_citations_shown_in_correct_order(
+    client: Client, alice: User, chat_history: ChatHistory, several_files: Sequence[File]
+):
+    # Given
+    client.force_login(alice)
+    chat_message = ChatMessage.objects.create(chat_history=chat_history, text="Some answer.", role=ChatRoleEnum.ai)
+
+    Citation.objects.create(file=several_files[1], chat_message=chat_message, text="Citation 1")
+    Citation.objects.create(file=several_files[0], chat_message=chat_message, text="Citation 2")
+    Citation.objects.create(file=several_files[1], chat_message=chat_message, text="Citation 3")
+    Citation.objects.create(file=several_files[2], chat_message=chat_message, text="Citation 4")
+    Citation.objects.create(file=several_files[0], chat_message=chat_message, text="Citation 5")
+
+    # When
+    response = client.get(f"/citations/{chat_message.id}/")
+
+    # Then
+    assert response.status_code == HTTPStatus.OK
+    soup = BeautifulSoup(response.content)
+    sources_panel = soup.select("div.iai-panel")[1]
+    files = sources_panel.find_all("h3")
```

Comment:
Further to our discussion in Slack, would we be able to use something like semantic class names to make this clearer (and possibly less brittle)?

Reply:
If @KevinEtchells is happy, so am I!